tests/Model/DataType/LinkTest.php

```diff
@@ -80,21 +80,4 @@ public function testSave()
         $this->assertEquals($link->getDirect(), $linkObjectReloaded->getTestlink()->getDirect());
         $this->assertEquals($link->getDirect(), $linkObjectReloaded->getLtestlink()->getDirect());
     }
-
-    /**
-     * Verifies that Link data throws correct exceptions if invalid data is given
-     *
-     * @throws \Exception
-     */
-    public function testCheckValidity()
-    {
-        $linkObject = $this->createLinkObject();
-        $linkObject->setTestlink('https://pimcore.com/');
-        $linkObject->setLtestlink('https://pimcore.com/');
-
-        $this->expectException(ValidationException::class);
-        $this->expectExceptionMessage('Expected DataObject\\Data\\Link or null');
```

Comment:
We can test if the TypeError will correct thrown.
```
        try {
            $linkObject = $this->createLinkObject();
            $linkObject->setTestlink('https://pimcore.com/');
            $linkObject->setLtestlink('https://pimcore.com/');
        } catch (\Throwable $e) {
            $this->assertInstanceOf(\TypeError::class, $e);
        }
```

Reply:
@blankse  Thank you I didn't now this workaround, but do you think we should test php behavior? Because the TypeError is not thrown by our code

Reply:
So we can test the correct typehint. The typehint is set by our code.

Reply:
Is also used in RgbaColor
https://github.com/pimcore/pimcore/blob/2ce8f9a7e2694b3a5b3f1b7e370dff150fe7cb8c/tests/_support/Helper/DataType/TestDataHelper.php#L1280-L1284

Reply:
Thanks for mentioning